docs/usage/types/dicts_mapping.md

```diff
@@ -18,3 +61,69 @@
     Prior to Python 3.8, it requires the [typing-extensions](https://pypi.org/project/typing-extensions/) package.
     But required and optional fields are properly differentiated only since Python 3.9.
     We therefore recommend using [typing-extensions](https://pypi.org/project/typing-extensions/) with Python 3.8 as well.
+
+Same as `dict` but Pydantic will validate the dictionary since keys are annotated.
```

Comment:
We need a better description of what a typeddict is here.

Reply:
Added some descriptions inspired by `typing.TypedDcit` doc